app/controllers/media/objects_controller.rb

```diff
@@ -1,79 +1,79 @@
 # frozen_string_literal: true
 
-class Media::ObjectsController < ApplicationController
-  before_action :set_media_object, only: %i[show delete]
-  skip_authorization_check
+module Media
+  # Creating, getting, and deleting media attached to responses.
+  class ObjectsController < ApplicationController
+    include Storage
 
-  def show
-    style = params[:style]
-    @answer = @media_object.answer
-    @response = @answer.try(:response)
-    disposition = params[:dl] == "1" ? "attachment" : "inline"
+    before_action :set_media_object, only: %i[show destroy]
+    skip_authorization_check
 
-    if @response
-      authorize!(:show, @response)
-    elsif @media_object.token != params[:token]
-      raise CanCan::AccessDenied.new("Not authorized", :view, :media_object)
-    end
+    def show
+      style = params[:style] || "original"
+      @answer = @media_object.answer
+      @response = @answer.try(:response)
+      disposition = params[:dl] == "1" ? "attachment" : "inline"
 
-    send_file(@media_object.item.path(style),
-      type: @media_object.item_content_type,
-      disposition: disposition,
-      filename: media_filename)
-  end
+      authorize!(:show, @response) if @response
+
+      send_attachment(
+        @media_object.item,
+        style: style,
+        type: @media_object.item_content_type,
```

Comment:
Don't think we are using this param anymore?

Reply:
I added the following to the storage concern: `params[:type] ||= attachment.content_type` which is essentially the same as what's happening here.  I'll remove this line from the objects controller.